test/jdk/java/net/httpclient/ContentLengthHeaderTest.java

```diff
@@ -0,0 +1,160 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Tests that a Content-length header is not sent for GET requests
+ *          that do not have a body. Also checks that the header is sent when
+ *          a body is present on the GET request.
+ * @bug 8283544
+ * @run testng/othervm ContentLengthHeaderTest
+ */
+
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import static java.net.http.HttpClient.Version.HTTP_1_1;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+
+
+public class ContentLengthHeaderTest {
+
+    final String GET_NO_BODY_PATH = "/get_no_body";
+    final String GET_BODY_PATH = "/get_body";
+    static final String GET_BODY = "Get with body";
+    static final String RESPONSE_BODY = "Test Response Body";
+
+    static HttpServer testContentLengthServer;
+    static PrintStream testLog = System.err;
+
+    String testContentLenURI;
+
+    @BeforeTest
+    public void setup() throws IOException {
+        InetSocketAddress sa = new InetSocketAddress(InetAddress.getLoopbackAddress(), 0);
+        testContentLengthServer = HttpServer.create(sa, 0);
+
+        // Create handlers for tests that check for the presence of a Content-length header
+        testContentLengthServer.createContext(GET_NO_BODY_PATH, new NoContentLengthHandler());
+        testContentLengthServer.createContext(GET_BODY_PATH, new ContentLengthHandler());
+        testContentLenURI = "http://" + testContentLengthServer.getAddress().getHostString() +
+                             ":" + testContentLengthServer.getAddress().getPort();
+
+        testContentLengthServer.start();
+        testLog.println("Server up at address: " + testContentLengthServer.getAddress());
+        testLog.println("Request URI for Client: " + testContentLenURI);
+    }
+
+    @AfterTest
+    public void teardown() {
+        testContentLengthServer.stop(0);
```

Comment:
Hello Conor, perhaps we should add a null check here for `testContentLengthServer` just in case the server creation failed in `setup()`.